test/e2e/funcs/feature.go

```diff
@@ -385,6 +390,72 @@ func DeleteResources(dir, pattern string) features.Func {
 	}
 }
 
+// CopyImageToRegistry tries to copy the supplied image to the supplied registry within the timeout
+func CopyImageToRegistry(clusterName, ns, sName, image string, timeout time.Duration) features.Func {
+	return func(ctx context.Context, t *testing.T, c *envconf.Config) context.Context {
+		reg, err := ServiceIngressEndPoint(ctx, c, clusterName, ns, sName)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		t.Logf("registry endpoint %s", reg)
+		srcRef, err := name.ParseReference(image)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		src, err := daemon.Image(srcRef)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		i := strings.Split(srcRef.String(), "/")
+		err = wait.For(func() (done bool, err error) {
+			err = crane.Push(src, fmt.Sprintf("%s/%s", reg, i[1]), crane.Insecure)
+			if err != nil {
+				return false, nil //nolint:nilerr // we want to retry and to throw error
+			}
+			return true, nil
+		}, wait.WithTimeout(timeout))
+		if err != nil {
+			t.Fatalf("copying image `%s` to registry `%s` not successful: %v", image, reg, err)
+		}
+
+		return ctx
+	}
+}
+
+// ClaimsManagedResourcesHaveLabel fails a test if the managed resources
+// created by the claim does not have the supplied value at the supplied label
+// path within the supplied duration.
+func ClaimsManagedResourcesHaveLabel(ns, apiVersion, kind, name, label, want string, timeout time.Duration) features.Func {
```

Comment:
nit: do you think it would be possible to make this more generic similar to `ResourcesHaveFieldValueWithin` below?
I would imagine something like:
```golang
func ClaimsManagedResourcesHaveFieldValueWithin(d time.Duration, ... something to specify the parent Claim ..., ... something to filter the managed resources ..., path string, want any) features.Func {
...
```
The other method is plural, so I kept it plural, but here the plural could be in the managed resources, instead of the claim 🤔 does it make sense?